packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -104,14 +100,16 @@ function evaluateStylesheetsContent(
 }
 
 export function getStylesheetsContent(vm: VM, template: Template): string[] {
-    const { stylesheets, stylesheetTokens: tokens } = template;
+    const { stylesheets, stylesheetTokens } = template;
     const { syntheticShadow } = vm.renderer;
 
     let content: string[] = [];
 
-    if (!isUndefined(stylesheets) && !isUndefined(tokens)) {
-        const hostSelector = syntheticShadow ? `[${tokens.hostAttribute}]` : '';
-        const shadowSelector = syntheticShadow ? `[${tokens.shadowAttribute}]` : '';
+    if (!isUndefined(stylesheets) && stylesheets.length !== 0) {
+        const tokens = syntheticShadow && hasShadow(vm) && stylesheetTokens;
```

Comment:
I had to pause and read multiple times this line to understand the implications of this logical expression. 

It would be great to add a comment explaining that we only want to add selector scoping when rendering a shadow DOM component with synthetic shadow enabled. For native shadow DOM and light DOM components no scoping is needed.

Reply:
Will do.